### PR TITLE
288 compare two maps side by side

### LINKED ISSUE #288: Compare two maps side by side (prototype)
Investigate how to do syncing Map zoom and scroll. 
Example see: https://openlayers.org/en/latest/examples/side-by-side.html
Need to investigate if it works for the layer switcher

### PR DESCRIPTION
**Pull request by ylyangtw**
- Add map comparison component. This should be reusable in other analyses
- Add Pole Damage comparison component.

### CODE REVIEW

File: src/utils/map.js [hunk 1]
```diff
@@ -138,3 +138,45 @@ export const validateCoord = (lon, lat, boundingBox) => {
 		parseFloat(lat) > boundingBox[3]
 	);
 };
+
+// Function to generate basicLayers objects
+export function createBasicLayers(title) {
```
Comment (longshuicy) [inline src/utils/map.js:143]: Don't we already have this utility function in the same mapUtil? 
What's the difference between this one and `export const basicLayers = new GroupLayer...`

**Review by longshuicy — COMMENTED** [review]
Side by side map works great! Good work~
There might be a few follow-ups on this PR you could do:

- [ ] Could you add the map component to all the compare widgets that have maps: 
  - [ ] Building damage
  - [ ] MCS, 
  - [ ] Functionality
  - [ ] Popoulation dislcoation I believe
- [ ] Could you work with @Rashmil-1999  to include the "clicking" building and get feature back to the comparison map as well?
I will tag you two to first review each other's PR and get familiar with the code.

File: src/utils/map.js (hunk 1 above)
Comment (ylyangtw) [reply src/utils/map.js:143]: Right. In openlayer, one layer can only be added to one map. The original `export const basicLayers` creates one object for basicLayers, which works fine for one map. But in comparison, we want two basicLayers being added to one map, so we will need to create two objects for basicLayers.

Here is how I do it in pole damage:
image

I have tried writing some cloning method in order to have less duplicated codes as possible. However, the sub tileLayers in basicLayers also need to be newly created... so I have no success how to do it better.

**Comment by ylyangtw** [comment]
> Side by side map works great! Good work~ There might be a few follow-ups on this PR you could do:
> 
> * [ ]  Could you add the map component to all the compare widgets that have maps:
>   
>   * [ ]  Building damage
>   * [ ]  MCS,
>   * [ ]  Functionality
>   * [ ]  Popoulation dislcoation I believe
> * [ ]  Could you work with @Rashmil-1999  to include the "clicking" building and get feature back to the comparison map as well?
>   I will tag you two to first review each other's PR and get familiar with the code.

Will address it another issues https://github.com/IN-CORE/incore-playbook/issues/317 and https://github.com/IN-CORE/incore-playbook/issues/320

**Comment by Rashmil-1999** [comment]
Wouldn't adding some space between the 2 maps look a bit better?

**Comment by ylyangtw** [comment]
> Wouldn't adding some space between the 2 maps look a bit better?

Good suggestion. I have added margin in between. Here is the updated snapshot.
image

File: src/utils/map.js (hunk 1 above)
Comment (longshuicy) [reply src/utils/map.js:143]: Ah understood.

**Review by longshuicy — APPROVED** [review]

**Pull request merged**